proxy/envoy/discovery.go

```diff
@@ -219,13 +222,16 @@ func NewDiscoveryService(ctl model.Controller, configCache model.ConfigStoreCach
 
 	// Flush cached discovery responses whenever services, service
 	// instances, or routing configuration changes.
-	serviceHandler := func(s *model.Service, e model.Event) { out.clearCache() }
-	if err := ctl.AppendServiceHandler(serviceHandler); err != nil {
-		return nil, err
-	}
-	instanceHandler := func(s *model.ServiceInstance, e model.Event) { out.clearCache() }
-	if err := ctl.AppendInstanceHandler(instanceHandler); err != nil {
-		return nil, err
+
+	if ctl != nil {
+		serviceHandler := func(s *model.Service, e model.Event) { out.clearCache() }
+		if err := ctl.AppendServiceHandler(serviceHandler); err != nil {
+			return nil, err
+		}
+		instanceHandler := func(s *model.ServiceInstance, e model.Event) { out.clearCache() }
+		if err := ctl.AppendInstanceHandler(instanceHandler); err != nil {
+			return nil, err
+		}
```

Comment:
Is ctl == nil an error case? If yes, check it in the beginning and return an error.

If not, is it worth logging it.